transit_odp/data_quality/pti/functions.py

```diff
@@ -232,14 +232,31 @@ def has_unregistered_service_codes(context, services):
     service_code_list = []
     service = services[0]
     ns = {"x": service.nsmap.get(None)}
+    flexible_service_code = None
 
     service_list = service.xpath("x:Service", namespaces=ns)
     for service in service_list:
+        flexible = service.xpath("x:ServiceClassification/x:Flexible", namespaces=ns)
+        flexible_service = service.xpath("x:FlexibleService", namespaces=ns)
+        if flexible or flexible_service:
+            flexible_service_code = service.xpath(
+                "string(x:ServiceCode)", namespaces=ns
+            )
         service_code_list.append(service.xpath("string(x:ServiceCode)", namespaces=ns))
     r = re.compile("[a-zA-Z]{2}\\d{7}:[a-zA-Z0-9]+$")
     registered_service_code = list(filter(r.match, service_code_list))
+    unregistered_service_code = list(
+        filter(lambda s: not r.match(s), service_code_list)
```

Comment:
Could we verify the pattern for unregistered service code instead of adding unregistered service code when the registered service code pattern check fails. This way we could avoid add service code to unregistered when an incorrect value is passed in the element